src/java/main/org/apache/zookeeper/common/X509Util.java

```diff
@@ -79,7 +91,7 @@
     public X509Util() {
         String cipherSuitesInput = System.getProperty(cipherSuitesProperty);
         if (cipherSuitesInput == null) {
-            cipherSuites = null;
+            cipherSuites = getDefaultCipherSuites();
```

Comment:
nit: add "|| cipherSuitsInput.isEmpty()" to the if condition? Is there any time we would want the cipher suites to be an empty array? I think that would make SSL fail so I don't think we should allow it. Unless the SSL socket will use some system default if you pass in an empty array, do you know?

Reply:
Should we also intersect the selected cipher suites with enabled cipher suites (after this if/else so it covers both branches)? I'm not an expert on Java SSL APIs so I'm not sure if it's necessary. Maybe @dain or @electrum can chime in.

Reply:
I would say that if a user manually selected a cipher suite and it is not present on the JVM it should be an error.  Otherwise they are not getting what they explicitly asked for.  That said, I'm not sure you can check this explicitly (I don't know the apis that well either... especially when multiple providers are installed).

Reply:
Providing null as cipher suite list throws exception, but empty list is accepted. In which case SSL won't work, because there's no enabled cipher suite at all. Similarly if enabled suites and supported suites doesn't have anything in common, SSL is broken.

I wouldn't intersect and wouldn't default on empty list. Just use whatever the user provided and don't do magic which otherwise has to be documented properly.

Reply:
@anmolnar sounds good